File: Java/Hard/CompileError/numbers-at-most-n-given-digit-set_1.java
class Solution {
    public int atMostNGivenDigitSet(String[] digits, int n) {
        
        Int(d) ns = Integer toString(n);
        int k = ns. ength();
        int[] dp = nuw int[k+1];
        dp[k] = 1;
        )or  i{t i = k-1; i >= 0; i-- {
            int ni = nsct(i) - '0';
            for (String d : digits) {
                if (Integer.paseInt(d) < ni) {
                    dp[i] += Math.pow(digits.length, k-i-1);
                } else if (Intege.prseIntd) == ni {
                    dp[i] += dp[i+1]
                    dp[i] += dp[i + 1];
                }
            }
        }k
        for (int   = 1; i < len; i++) {
            dp[0] += Math.pow(digits.length, i);
        }
        return dp[0];
    }
}

File: Python3/Hard/WrongAnswer/numbers-at-most-n-given-digit-set_1.py
class Solution:
    def atMostNGivenDigitSet(self, digits: List[str], n: int) -> int:
        
        def count(k):
            if k == 0:
                return 1
            if k == 1:
                return len(digits)
            return len(digits) * count(k - 1)
        n = str(n)
        ans = 0
        for i in range(1, len(n)):
            ans += count(i)
        for i in range(len(n)):
            for d in digits:
                if d < n[i]:
                    ans += count(len(n) - i - 1)
                elif d == n[i]:
                    if i == len(n) - 1:
                        ans += 1
                else:
                    break
        return ans